### PR TITLE
Python312

### PR DESCRIPTION
**Pull request by keileg**
## Proposed changes
This PR introduces the updates necessary to run PorePy on Python 3.12.

For now, we keep Python 3.10-12, and run some of the checks on 3.11 and 3.12.

For reference, only minor changes was necessary to make the code run.

The PR should not be merged until pytest has been run including skipped tests.

## Types of changes

What types of changes does this PR introduce to PorePy?
_Put an `x` in the boxes that apply._

- [ ] Minor change (e.g., dependency bumps, broken links).
- [ ] Bugfix (non-breaking change which fixes an issue).
- [ ] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).
- [ ] Testing (contribution related to testing of existing or new functionality).
- [ ] Documentation (contribution related to adding, improving, or fixing documentation).
- [x] Maintenance (e.g., improve logic and performance, remove obsolete code).
- [ ] Other:

## Checklist

_Put an `x` in the boxes that apply or explain briefly why the box is not relevant._

- [x] The documentation is up-to-date.
- [x] Static typing is included in the update.
- [x] This PR does not duplicate existing functionality.
- [x] The update is covered by the test suite (including tests added in the PR).
- [] If new skipped tests have been introduced in this PR, `pytest` was run with the `--run-skipped` flag.

### CODE REVIEW

**Comment by keileg** [comment]
All tests pass.

File: .github/workflows/run-pytest-all.yml [hunk 1]
```diff
@@ -21,7 +21,7 @@ jobs:
     # https://help.github.com/en/actions/language-and-framework-guides/using-python-with-github-actions#specifying-a-python-version
     strategy:
       matrix:
-        python-version:  ["3.11"]  # run only the last python version
+        python-version:  ["3.11", "3.12"]  # run only the last python version
```
Comment (IvarStefansson) [inline .github/workflows/run-pytest-all.yml:24]: Comment or code should be updated

File: .github/workflows/run-static-checks.yml [hunk 2]
```diff
@@ -31,7 +31,7 @@ jobs:
     strategy:
       matrix:
         # Run the tests only on the latest supported Python version (currently 3.10).
```
Comment (IvarStefansson) [inline .github/workflows/run-static-checks.yml:33]: Please update comment, possibly also specs in next line

File: src/porepy/fracs/split_grid.py [hunk 3]
```diff
@@ -1051,7 +1051,7 @@ def _sort_sub_list(
 
 
 def _find_cell_color(g: pp.Grid, cells: np.ndarray) -> np.ndarray:
-    """Color the cells depending on the cell connections.
+    r"""Color the cells depending on the cell connections.
 
     Each group of cells that are connected (either directly by a shared face or
     through a series of shared faces of many cells) is are given different colors.
```
Comment (IvarStefansson) [inline src/porepy/fracs/split_grid.py:1057]:
```suggestion
    through a series of shared faces of many cells) is given a distinct color.
```

**Review by IvarStefansson — APPROVED** [review]
Minor stuff only.

**Pull request merged**